q2_quality_control/decontam.py

```diff
@@ -0,0 +1,163 @@
+    'metadata': _SKIP,
+    'threshold': _PER_NUM,
+    'method': _DECON_METHOD_STR,
+    'freq_concentration_column': _SKIP,
+    'prev_control_column': _SKIP,
+    'prev_control_indicator': _SKIP,
+    'decontam_scores': _SKIP,
+}
+
+
+def _check_inputs(**kwargs):
+    for param, arg in kwargs.items():
+        check_is_valid, explanation = _valid_inputs[param]
+        if not check_is_valid(arg):
+            raise ValueError('Argument to %r was %r, should be %s.'
+                             % (param, arg, explanation))
+
+
+def _check_column_inputs(metadata, method, freq_concentration_column,
+                         prev_control_column, prev_control_indicator):
+    if method == 'prevalence':
+        if prev_control_column not in metadata.columns:
+            raise ValueError('Prevalence column not found, please '
+                             'select from:\n'
+                             + str(', '.join(metadata.columns)))
+        else:
+            if prev_control_indicator not in list(
+                    metadata[prev_control_column]):
+                raise ValueError('No control values found, please select '
+                                 'from:\n' +
+                                 str(', '.join(
+                                     metadata[prev_control_column]
+                                     .unique())))
+            else:
+                print("All inputs are found")
+    elif method == 'frequency':
+        if freq_concentration_column not in metadata.columns:
+            raise ValueError('Frequency column not found, please '
+                             'select from:\n'
+                             + str(', '.join(metadata.columns)))
+        else:
+            print("All inputs are found")
+    else:
+        if ((freq_concentration_column not in metadata.columns) or
+                (prev_control_column not in metadata.columns)):
+            raise ValueError('Column id input error, please '
+                             'select from:\n'
+                             + str(', '.join(metadata.columns)))
+        else:
+            if prev_control_indicator not in list(
+                    metadata[prev_control_column]):
+                raise ValueError('No control values found, please select '
+                                 'from:\n' +
+                                 str(', '.join(
+                                     metadata[prev_control_column]
+                                     .unique())))
+            else:
+                print("All inputs are found")
+
+
+def _decontam_identify_helper(track_fp, method):
+    df = pd.read_csv(track_fp, sep='\t', index_col=0)
+    df.index.name = '#OTU ID'
+    # removes last column containing true/false information from the dataframe
+    df = df.drop(df.columns[(len(df.columns)-1)], axis=1)
+    if method == 'combined':
+        df = df.fillna(0)
+
+    # removes all columns that are completely empty
+    temp_transposed_table = df.transpose()
```

Comment:
This transpose (and the "closing" transpose below this) can be removed if `dropna()` is given `axis='columns'`.